src/main/java/com/hbm/world/gen/component/Component.java

```diff
@@ -663,7 +673,44 @@ protected void fillWithBlocks(World world, StructureBoundingBox box, int minX, i
 			}
 		}
 	}
-	
+	/** Have you ever wanted Glyphids in your building? No? Too bad! **/
+	protected void infest(World world, StructureBoundingBox box, int minX, int minY, int minZ, int maxX, int maxY, int maxZ, int maxSize, Random rand){
+		if(!enableInfestation)
+			return;
+
+		PollutionHandler.PollutionPerWorld ppw = PollutionHandler.perWorld.get(world);
+
+		int pollutedLocations = ppw == null ? 0 : ppw.pollution.size();
```

Comment:
Treating the pollution system as a global thing doesn't really make sense since pollution doesn't even spread this far. Plus having it work like that is going to disproportionally affect servers, as there would be multiple bases with more pollution sources, even if they are really spread out.

Reply:
Hmm, fair enough, this thing alone caused a lot of headache when testing